### PR TITLE
Add ml-bindport / METALLB_ML_BIND_PORT config

### LINKED ISSUE #581: Installation conflict with Docker EE
There is an installation conflict when applying the MetalLB manifest on a Kubernetes cluster that is part of a Docker EE server cluster.  The error is a port conflict with Swarm.  Here are the error logs generated:

`$ kubectl logs pod/speaker-k2wq4 -n metallb-system
{"branch":"main","caller":"main.go:82","commit":"9d6de1ee","msg":"MetalLB speaker starting (commit 9d6de1ee, branch main)","ts":"2020-04-14T01:08:29.480424823Z","version":""}
{"caller":"main.go:202","component":"MemberList","msg":"memberlist.go:145: [DEBUG] memberlist: Got bind error: Failed to start TCP listener on \"10.110.57.13\" port 7946: listen tcp 10.110.57.13:7946: bind: address already in use","ts":"2020-04-14T01:08:29.480944095Z"}
{"caller":"main.go:118","error":"Could not set up network transport: failed to obtain an address: Failed to start TCP listener on \"10.110.57.13\" port 7946: listen tcp 10.110.57.13:7946: bind: address already in use","msg":"failed to create memberlist","op":"startup","ts":"2020-04-14T01:08:29.480975464Z"}
`

I was able to overcome this by removing the METALLB_ML_BIND_ADDR environment variable from the speaker DaemonSet.

     - Version of MetalLB: 0.9.3
     - Version of Kubernetes: v1.14.8-docker-1
     - Name and version of network addon (e.g. Calico, Weave...): Calico v3.8.2
     - Whether you've configured kube-proxy for iptables or ipvs mode: N/A

### PR DESCRIPTION
**Pull request by champtar**
Docker EE already uses port 7946 (likely memberlist)
Fixes #581

### CODE REVIEW

File: speaker/main.go [hunk 1]
```diff
@@ -71,6 +72,7 @@ func main() {
 		config      = flag.String("config", "config", "Kubernetes ConfigMap containing MetalLB's configuration")
 		host        = flag.String("host", os.Getenv("METALLB_HOST"), "HTTP host address")
 		mlBindAddr  = flag.String("ml-bindaddr", os.Getenv("METALLB_ML_BIND_ADDR"), "Bind addr for MemberList (fast dead node detection)")
+		mlBindPort  = flag.String("ml-bindport", os.Getenv("METALLB_ML_BIND_PORT"), "Bind port for MemberList (fast dead node detection)")
```
[the remark this answers is not in the document]
Comment (champtar) [reply speaker/main.go:75]: I agree with the second part, it's future proof like that

File: speaker/main.go [hunk 2]
```diff
@@ -106,6 +108,14 @@ func main() {
 		// mconfig.Name MUST be spec.nodeName, as we will match it against Enpoints nodeName in usableNodes()
 		mconfig.Name = *myNode
 		mconfig.BindAddr = *mlBindAddr
+		if *mlBindPort != "" {
+			port, err := strconv.Atoi(*mlBindPort)
+			if err != nil {
+				logger.Log("op", "startup", "error", "unable to parse ml-bindport", "msg", err)
```
[the remark this answers is not in the document]
Comment (champtar) [reply speaker/main.go:114]: I though about it ... and forgot

File: speaker/main.go [hunk 3]
```diff
@@ -106,6 +108,14 @@ func main() {
 		// mconfig.Name MUST be spec.nodeName, as we will match it against Enpoints nodeName in usableNodes()
 		mconfig.Name = *myNode
 		mconfig.BindAddr = *mlBindAddr
+		if *mlBindPort != "" {
+			port, err := strconv.Atoi(*mlBindPort)
```
[the remark this answers is not in the document]
Comment (champtar) [reply speaker/main.go:112]: thanks !!

File: speaker/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rata) [reply speaker/main.go:75]: Cool, let's leave it like that, then. Thanks :)

**Comment by champtar** [comment]
@rata done

**Comment by champtar** [comment]
nop, not tested yet

**Comment by rata** [comment]
@champtar please ping me when it's ready to merge (if you don't have permissions) :)

**Comment by champtar** [comment]
Just tested, it works !
`docker.io/champtar/metallbspeaker:c3dfa2aa8113ee868edbad527e7f4fbab2d347ca`
Do you want me to add the port configuration commented in the manifest ?

**Comment by rata** [comment]
@champtar cool! Yes, adding it to the manifest commented out (maybe a comment explaining and saying it is not usually needed or something?) sounds great! :)

**Pull request merged**